invenio_app_ils/circulation/mail/tasks.py

```diff
@@ -12,59 +12,74 @@
 from flask import current_app
 from invenio_circulation.proxies import current_circulation
 
-from invenio_app_ils.circulation.mail.factory import loan_message_factory
+from invenio_app_ils.circulation.mail.factory import \
+    loan_message_creator_factory
+from invenio_app_ils.circulation.search import get_all_expiring_loans, \
+    get_all_overdue_loans
 from invenio_app_ils.circulation.utils import circulation_overdue_loan_days
-from invenio_app_ils.documents.api import Document
+from invenio_app_ils.mail.messages import get_common_message_ctx
 from invenio_app_ils.mail.tasks import send_ils_email
-from invenio_app_ils.records.api import Patron
 
 celery_logger = get_task_logger(__name__)
 
 
-def send_loan_mail(trigger, loan, message_ctx={}, **kwargs):
+def send_loan_mail(loan_action, loan, message_ctx={}, **kwargs):
     """Send loan email message asynchronously and log the result in Celery.
 
-    :param trigger: Loan action trigger.
-    :param loan: Updated loan.
+    :param loan_action: the triggered loan action.
+    :param loan: the loan record.
     :param message_ctx: any other parameter to be passed as ctx in the msg.
     """
-    patron = Patron.get_patron(loan["patron_pid"])
-    document = Document.get_record_by_pid(loan["document_pid"])
-    factory = loan_message_factory()
-    msg = factory(
-        trigger,
-        message_ctx=dict(
-            loan=loan,
-            document=dict(title=document["title"]),
-            patron=patron,
-            **message_ctx,
-        ),
+    creator = loan_message_creator_factory()
+
+    message_ctx.update(get_common_message_ctx(record=loan))
+    patron = message_ctx["patron"]
+
+    msg = creator(
+        loan,
+        action=loan_action,
+        message_ctx=message_ctx,
         recipients=[patron.email],
         **kwargs,
     )
     send_ils_email(msg)
 
 
-def send_loan_overdue_reminder_mail(loan):
-    """Send loan overdue email message async and log the result in Celery.
-
-    :param loan: the overdue loan.
-    """
+def send_loan_overdue_reminder_mail(loan, days_ago):
+    """Send loan overdue email."""
     send_loan_mail(
-        "overdue_reminder",
+        loan_action="overdue_reminder",
```

Comment:
minor: In relation to my previous comment, for `reason` to send email, or it could be `email_type`, `email_reason`, `email_condition`... or something else